containers/wildfly/src/main/java/org/jboss/qa/jcontainer/wildfly/WildflyConfiguration.java

```diff
@@ -177,33 +178,38 @@ public WildflyConfiguration build() {
 			}
 
 			// Set JAVA_OPTS
-			final StringBuilder allJavaOpts = javaOpts == null ? new StringBuilder() : javaOpts;
+			javaOpts = javaOpts.stream().map(i -> i.replaceAll("\\s","")).collect(Collectors.toList());
 			if (!StringUtils.isEmpty(xms)) {
-				allJavaOpts.append(" -Xms").append(xms);
+				replaceJavaOptIfExists("-Xms", xms);
 			}
 			if (!StringUtils.isEmpty(xmx)) {
-				allJavaOpts.append(" -Xmx").append(xmx);
+				replaceJavaOptIfExists("-Xmx", xmx);
 			}
 			if (!StringUtils.isEmpty(permSize)) {
-				allJavaOpts.append(" -XX:PermSize=").append(permSize);
+				replaceJavaOptIfExists("-XX:PermSize=", permSize);
 			}
 			if (!StringUtils.isEmpty(maxPermSize)) {
-				allJavaOpts.append(" -XX:MaxPermSize=").append(maxPermSize);
+				replaceJavaOptIfExists("-XX:MaxPermSize=", maxPermSize);
 			}
-			allJavaOpts.append(" -Djava.net.preferIPv4Stack=true");
-			allJavaOpts.append(" -Djboss.modules.system.pkgs=org.jboss.byteman");
-			allJavaOpts.append(" -Djava.awt.headless=true");
-			allJavaOpts.append(" -Djboss.socket.binding.port-offset=").append(portOffset);
+			replaceJavaOptIfExists("-Djava.net.preferIPv4Stack=", "true");
```

Comment:
dont replace users configuration for `-Djava.net.preferIPv4Stack=`, `-Djboss.modules.system.pkgs="` and `-Djava.awt.headless=`. Let users to overflow default configuration. For example I want to add multiply system pkgs like `-Djboss.modules.system.pkgs=org.jboss.byteman,my.package`